awx/main/tasks/jobs.py

```diff
@@ -1199,21 +1040,13 @@ class RunProjectUpdate(BaseTask):
 
     model = ProjectUpdate
     event_model = ProjectUpdateEvent
-    event_data_key = 'project_update_id'
+    callback_class = RunnerCallbackForProjectUpdate
 
     def __init__(self, *args, job_private_data_dir=None, **kwargs):
         super(RunProjectUpdate, self).__init__(*args, **kwargs)
-        self.playbook_new_revision = None
         self.original_branch = None
         self.job_private_data_dir = job_private_data_dir
-
-    def event_handler(self, event_data):
-        super(RunProjectUpdate, self).event_handler(event_data)
-        returned_data = event_data.get('event_data', {})
-        if returned_data.get('task_action', '') == 'set_fact':
-            returned_facts = returned_data.get('res', {}).get('ansible_facts', {})
-            if 'scm_version' in returned_facts:
-                self.playbook_new_revision = returned_facts['scm_version']
+        self.runner_callback = self.callback_class()
```

Comment:
You shouldn't need this line, because it is done in `super(RunProjectUpdate, self).__init__(*args, **kwargs)`. That runs the `BaseTask` version of the init method.

When `BaseTask.__init__` runs, it will do this, `self.runner_callback = self.callback_class()`, and when it does that `self` is an instance of `RunProjectUpdate`, so it'll pick up the class property of `callback_class` so it should correctly make the project update version of the callback object.